src/sentry/integrations/base.py

```diff
@@ -463,3 +468,82 @@ def notify_remove_external_team(self, external_team: ExternalActor, team: Team)
 
     def remove_notification_settings(self, actor_id: int, provider: str) -> None:
         pass
+
+
+def is_response_success(resp) -> bool:
+    if resp.status_code:
+        if resp.status_code < 300:
+            return True
+    return False
+
+
+def is_response_error(resp) -> bool:
+    if resp.status_code:
+        if resp.status_code >= 400 and resp.status_code != 429 and resp.status_code < 500:
+            return True
+    return False
+
+
+def disable_integration(
+    buffer: IntegrationRequestBuffer, redis_key: str, integration_id: int
+) -> None:
+    from sentry.services.hybrid_cloud.integration import integration_service
+
+    result = integration_service.organization_contexts(integration_id=integration_id)
+    rpc_integration = result.integration
+    rpc_org_integrations = result.organization_integrations
+    if rpc_integration and rpc_integration.status == ObjectStatus.DISABLED:
+        return
+
+    org = None
+    if len(rpc_org_integrations) > 0:
+        org_context = organization_service.get_organization_by_id(
+            id=rpc_org_integrations[0].organization_id,
+            include_projects=False,
+            include_teams=False,
+        )
+        if org_context:
+            org = org_context.organization
+
+    extra = {
+        "integration_id": integration_id,
+        "buffer_record": buffer._get_all_from_buffer(),
+    }
+    if len(rpc_org_integrations) == 0 and rpc_integration is None:
+        extra["provider"] = "unknown"
+        extra["organization_id"] = "unknown"
+    elif len(rpc_org_integrations) == 0:
+        extra["provider"] = rpc_integration.provider
+        extra["organization_id"] = "unknown"
+    elif rpc_integration is None:
+        extra["provider"] = "unknown"
+        extra["organization_id"] = rpc_org_integrations[0].organization_id
+    else:
+        extra["provider"] = rpc_integration.provider
+        extra["organization_id"] = rpc_org_integrations[0].organization_id
```

Comment:
can this be simplified?
```suggestion
extra["provider"] = "unknown" if rpc_integration is None else rpc_integration.provider
extra["organization_id"] = "unknown" if len(rpc_org_integrations) == 0 else rpc_org_integrations[0].organization_id
```